### PR TITLE
Allow the project to be mapped to a location other than //flutter.

### PR DESCRIPTION
**Pull request by pylaligand**
This is for Fuchsia where we would like it to be located at //third_party/flutter.

### CODE REVIEW

**Comment by pylaligand** [comment]
/cc @abarth @Hixie @chinmaygarde 

This is a proposal that would allow us to nest the Flutter engine under `//third_party` in Fuchsia. I'm certain more needs to be done in this project but I was able to build all of Fuchsia with this by simply adjusting some paths and modifying an SDK-related script under `//dart`. The value for the `flutter` variable should be set in `//build/config/BUILDCONFIG.gn`, to `"flutter"` in your case and `"third_party/flutter"` in ours.

Comments welcome!

File: BUILD.gn [hunk 1]
```diff
@@ -6,56 +6,62 @@ group("flutter") {
...
+      "//$flutter/sky/engine/wtf:wtf_unittests",
+      "//$flutter/synchronization:synchronization_unittests",
+      "//$flutter/third_party/txt:txt_benchmarks",
+      "//$flutter/third_party/txt:txt_unittests",
       "//garnet/public/lib/fxl:fxl_unittests",
     ]
   }
 }
 
+config("config") {
```
Comment (pylaligand) [inline BUILD.gn]: I had to define this config and add it to various `source_set`s for includes to work.

File: BUILD.gn [hunk 2]
```diff
@@ -6,56 +6,62 @@ group("flutter") {
...
 
   deps = [
-    "//flutter/lib/snapshot:compile_platform",
-    "//flutter/lib/snapshot:generate_snapshot_bin",
-    "//flutter/sky",
-    "//flutter/third_party/txt",
+    "//$flutter/lib/snapshot:compile_platform",
+    "//$flutter/lib/snapshot:generate_snapshot_bin",
+    "//$flutter/sky",
+    "//$flutter/third_party/txt",
```
Comment (abarth) [inline BUILD.gn]: The normal idiom is to put the `//` inside the variable, so this would read `$flutter/third_party/txt`

Comment (pylaligand) [reply BUILD.gn]: Done.

**Comment by pylaligand** [comment]
Btw I'm aware Travis is complaining, but I'd like to have your opinion on the approach before making changes to the buildroot project.

**Comment by chinmaygarde** [comment]
Instead of polluting the global GN namespace with Flutter specific variables, we can just commit to making all references to the Flutter specific libraries be relative to engine root instead of the build root. I don't see why Flutter should care about where in the build root it is located.

**Comment by pylaligand** [comment]
Do you mean that instead of `shell/platform/darwin/BUILD.gn` referring to `//flutter/shell/gpu` it should use `../../gpu` or `../../../shell/gpu`?

**Comment by chinmaygarde** [comment]
> Do you mean that instead of `shell/platform/darwin/BUILD.gn` referring to `//flutter/shell/gpu` it should use `../../gpu` or `../../../shell/gpu`?

Yeah. But now that I see it actually written out, it does seem a bit inelegant. I'd still prefer not to pollute the global namespace though. WDYT?

**Comment by pylaligand** [comment]
Agreed on the lack of elegance. I'm running out of ideas though. IMO the extra variable is not that big a deal, here's what I had to do for Fuchsia's build: https://fuchsia-review.googlesource.com/#/c/build/+/74442/.

**Comment by pylaligand** [comment]
`flutter_root` it is!

**Pull request merged**